### DISCUSSION TITLE
Coravel responding differently to a published app over dotnet run

### DISCUSSION THREAD

**Discussion by mhegreberg** [jamesmh/coravel#309] [Q&A] [answered]
I have a worker app that performs some notifications and checks for my webapp. When I run the app using `dotnet run`, all the services seem to start on schedule, and run fine, but after publishing the app, it doesn't even try to run them. The app will start, and then just sit silently forever.

This is on .NET 6. here is the snippet from `Program.cs` that schedules everything
```
    IHost host = Host.CreateDefaultBuilder(args)
        .UseSerilog()
        .UseWindowsService(options => // this allows the app to be run as a windows service, but doesn't impede running on linux
                {
                    options.ServiceName = "DoNUTS Worker";
                })
        .ConfigureServices((ctx, services) =>
        {
            //services.AddHostedService<Worker>();
            services.AddScheduler();
            services.AddTransient<CheckLicenses>();
            services.AddTransient<CheckReminders>();
            services.AddTransient<SendMessages>();
            services.AddDbContext<InventoryContext>(options =>
                options.UseSqlServer(ctx.Configuration.GetConnectionString("InventoryContext"))); //get from user secrets or env. variables
        })
        .Build();


    host.Services.UseScheduler(scheduler =>

            {
                var jobschedule = scheduler.Schedule<CheckLicenses>();
                jobschedule.DailyAtHour(7)
                           .RunOnceAtStart();
            });
    host.Services.UseScheduler(scheduler =>

            {
                var jobschedule = scheduler.Schedule<CheckReminders>();
                jobschedule.DailyAt(7, 15)
                           .RunOnceAtStart();
            });
    host.Services.UseScheduler(scheduler =>

            {
                var jobschedule = scheduler.Schedule<SendMessages>();
                jobschedule.EveryThirtySeconds()
                .PreventOverlapping("SendMail");
            });


	Log.Information("Setup Complete");
    await host.RunAsync();
```

I feel like I'm going crazy. I've never had an app like this respond differently after being published. I'm sure I'm missing something simple.

**Comment by jamesmh** [comment]
This looks okay at a glance... I've done this successfully and I know others have. Might be something missing in the publishing process? It's a bit involved -> https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-6.0&tabs=visual-studio&viewFallbackFrom=aspnetcore-3.0#app-configuration

Let me know if you still need help 👍

**Reply by mhegreberg** [reply]
Is this still necessary if I'm just deploying it as an exe? 
I hadn't yet gotten around to setting it up as a windows service yet because running the .exe on it's own didn't run the scheduled tasks.

I'll go ahead and walk through this guide and see if it helps!

**Reply by mhegreberg** [reply]
I've removed everything Related to a windows service, and am still seeing a similar situation running this code from windows.
(As a note, I do my development from linux in WSL, but my production env. is mostly windows servers)
I've created the simplest case in a new repo: https://github.com/mhegreberg/CoravelBasicExample

when running this using `dotnet run` on linux, I get the `Starting Program` and the `Task Invoked!`, which tells me it ran the scheduled task

when I run `dotnet publish -r win-x64 --self-contained` , then run the exe from windows, I only get the `Starting Program`, so no invoked task.

Interestingly in this case, a published file works on linux, where it didn't on my full application. In Both cases Coravel does not seem to be running scheduled tasks from a windows environment.

I hope this basic case is helpful. I'm not sure what I'm missing here.

**Comment by mhegreberg** [comment, marked as answer]
I was able to resolve this.
I had a configuration issue with how I read in my environment variables. This caused the app to silently hang once a task was launched. I didn't notice because I was getting my user secrets in development, and the logging env. variables were loading properly.


I can now run it on linux and windows, and start it at a service.

Thanks!

**Comment by jamesmh** [comment]
Great! Glad to hear things are working 🙂
